### DISCUSSION TITLE
doublerulesep and vlines color

### DISCUSSION THREAD

**Discussion by libinxu01** [lvjr/tabularray#379] [Q&A]
I want a table with 4 colorful borders around. The up and below borders use a merged cell, and the left and right borders use double rule(because I find that a 2.5mm width column is not accurate). But I can't find the way to set the doublerulesep color and vline{2-Y}={1,Z}{0.5pt,fg=bgColorFourColorRed} doesn't work. I wonder how to achieve it.
![q](https://user-images.githubusercontent.com/15907992/224493273-7da9fefe-87fa-42e7-9884-935113e35fb8.png)

```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xespotcolor}
\usepackage{xcolor}
\usepackage{changepage}
\definecolor{bgColorFourColorRed}   {cmyk}{0,    0.11,  0.07,   0}

\NewTblrTheme{fancy}{
    \SetTblrTemplate{head,foot}{empty}
}

\begin{document}
	\begingroup
	\begin{adjustwidth}{-10em}{-10em}
		\begin{center}
			\begin{longtblr}[
				theme = fancy, 
				]{
					width=\dimexpr\textwidth\relax + 5mm, 
					colspec={||X[3,l,m]X[2,l,m]||}, 
					rulesep=2.5mm,
					row{1,Z}={bg=bgColorFourColorRed}, 
				hline{1, Z}={0.5pt,fg=bgColorFourColorRed}, 
				hline{2-Y}={2,Y}{0.5pt,fg=black,leftpos=0, rightpos=0}, 
                    vline{1,Z}={0.5pt,fg=bgColorFourColorRed}, 
				vline{2-Y}={1,Z}{0.5pt,fg=bgColorFourColorRed}, 
                    vline{2-Y}={2-Y}{0.5pt,fg=black}, 
                    rowhead=1,
                    rowfoot=1
				}
                    \SetCell[c=2]{c}{} & \\
				a & b \\
				a & b \\
				a & b \\
                    \SetCell[c=2]{c}{} & \\
                \end{longtblr}
		\end{center}
	\end{adjustwidth}
	\endgroup
	\par
 
	\begingroup
	\begin{adjustwidth}{-10em}{-10em}
		\begin{center}
                 \begin{longtblr}[
				theme = fancy, 
				]{
					width=2.5mm, 
					hlines,
                    vlines
				}
                    \SetCell[c=1]{c}{} \\
                \end{longtblr}
            \end{center}
        \end{adjustwidth}
	\endgroup
	\par
\end{document}

```

**Comment by muzimuzhi** [comment]
> ... and the left and right borders use double rule(because I find that a 2.5mm width column is not accurate).

An example with first and last columns having exact 2.5mm width:
```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xcolor}

\begin{document}
\def\testCellWidth#1{%
  \hskip-\leftsep\rlap{\color{red}\rule{#1}{3pt}}
}
\begin{longtblr}[label=none]{
  hlines, vlines,
  colspec={Q[wd=2.5mm] X X Q[wd=2.5mm]},
  column{1,Z}={colsep=0pt},
}
  \testCellWidth{2.5mm} & a & b & \testCellWidth{2.5mm} \\
  & a & b & \\
\end{longtblr}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/224505123-7e82fd27-0985-41cc-b28a-97ae684cfaab.png)

BTW I don't understand what's the `\begingroup ... \endgroup` are for since most environments changes states locally, what's the `center` envs are for since `longtblr` are automatically centered.

The `width=2.5mm` in your second `longtblr` has no effect because

> Option `width` is for setting the width of the table with extendable columns.
> c.f. `texdoc tabularray` 2023A, sec. 2.6.1 _Colspec and Width_

and what you want is to set width for columns, not the whole table.

Last
- It seems as of 2023A `tabularray` provides no way to set color for separation between tabular rules, see #287.
- Actually what you want is a tabular frame, which has no support on `tabularray`'s side yet. You could file a feature request issue for it.

**Reply by muzimuzhi** [reply]
A revised example
```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xcolor}

\begin{document}
\def\testCellWidth#1{%
  \hskip-\leftsep\rlap{\color{red}\rule{#1}{3pt}}
}

% trick, set width for all rules. just for test
\setlength\rulewidth{3pt}
\begin{longtblr}[label=none]{
  colspec={Q[wd=2.5mm, bg=red!10] X X Q[wd=2.5mm, bg=red!10]},
  column{1,Z}={colsep=0pt},
  row{1,Z}={bg=red!10},
  vline{2-Y}={2-Y}{solid},
  vline{2-Y}={1,Z}{solid, fg=red!10},
  hline{2-Y}={2-Y}{solid},
  hline{2-Y}={1,Z}{solid, fg=red!10, leftpos=0},
}
  \\
  \testCellWidth{2.5mm} & a & b & \testCellWidth{2.5mm}\\
  & a & b & \\
  \\
\end{longtblr}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/224505947-c2e97239-2bd0-4f24-a172-bd7d4fa99129.png)

**Reply by libinxu01** [reply]
Thank you so much!
